src/basic/virt.c

```diff
@@ -545,7 +550,12 @@ Virtualization detect_vm(void) {
                 return v;
 
 finish:
-        if (v == VIRTUALIZATION_NONE && other)
+        /* None of the checks above gave us a clear answer, hence let's now use fallback logic: if hyperv
+         * enlightenments are available but the VMM wasn't recognized as anything yet, it's probably
+         * Microsoft. */
+        if (IN_SET(v, VIRTUALIZATION_NONE, VIRTUALIZATION_VM_OTHER) && hyperv)
```

Comment:
Hmm, there's no need to check for `VIRTUALIZATION_VM_OTHER` here? I.e. it's unreachable as `detect_vm_zvm` would not return it, and in previous methods VIRTUALIZATION_VM_OTHER makes the search carry on.